s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -63,9 +63,12 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
   // def because tokens can expire
   def signingKey = SigningKey(settings.credentialsProvider, CredentialScope(LocalDate.now(), settings.s3Region, "s3"))
 
-  def download(s3Location: S3Location, range: Option[ByteRange] = None): Source[ByteString, Future[ObjectMetadata]] = {
+  def download(s3Location: S3Location,
+               range: Option[ByteRange] = None,
+               sse: Option[ServerSideEncryption] = None): Source[ByteString, Future[ObjectMetadata]] = {
```

Comment:
As this is internal API I'd prefer to not use default parameters (applies to several methods).

Reply:
@ennru Are you proposing simply remove the default parameters or split this method in 4 methods?

Reply:
All parameters (without defaults) in a single method to keep the internal API smaller.